### DISCUSSION TITLE
"Play Button" is not displayed in the top right corner of the editor.

### DISCUSSION THREAD

**Discussion by DataSailor** [microsoft/vscode-cpptools#12139] [Q&A] [answered]
According to the document tutorial, when I need to compile a C++ file, I should find the "Play Button" in the top right corner of the editor and select "Run C/C++ File". But I can't find this button now.

<img width="1132" alt="image" src="https://github.com/microsoft/vscode-cpptools/assets/39119487/05140842-170d-4a44-a667-f68b93d7acf5">

I found the option `C_Cpp: Debug Shortcut` in the settings, which seems to be related. However, I have already checked it, but the corresponding button did not appear.

How do I make the "Play Button" appear, or how do I compile C++ files?

Thank you for your answer.

**Comment by sean-mcmanus** [comment]
@DataSailor What version of VS Code are you using? What language is shown in the status bar in the bottom right? Does running the C/C++: Log Diagnostics command show output?

**Reply by DataSailor** [reply]
The environment versions I am using are MacOS 14.4.1, VSCode 1.87.2, and C/C++ 1.19.9.

The language used is displayed in the bottom right corner as `C++`.

I found a very strange problem when I use `Cmd+Shift+P` and search for `C/C++` related commands, the list only contains three options.

<img width="643" alt="image" src="https://github.com/microsoft/vscode-cpptools/assets/39119487/fead8436-aa02-4e9a-aa91-d3bd51b12d8e">

Therefore, I cannot find and running `C/C++: Log Diagnostics`

**Reply by sean-mcmanus** [reply, marked as answer]
@DataSailor I believe it's caused by C_Cpp.intelliSenseEngine set to "disabled", see https://github.com/microsoft/vscode-cpptools/issues/11909

**Reply by DataSailor** [reply]
When I restore the `C_Cpp.intelliSenseEngine` option to its default settings, the button appears normally. Thank you!
